src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -330,170 +344,186 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
+                                .getTransactionId()
+                )
+                .payments(
+                        transactionDocument
+                                .getPaymentNotices()
+                                .stream()
+                                .map(
+                                        paymentNotice -> new PaymentInfoDto()
+                                                .amount(paymentNotice.getAmount())
+                                                .reason(paymentNotice.getDescription())
+                                                .paymentToken(paymentNotice.getPaymentToken())
+                                                .rptId(paymentNotice.getRptId())
+                                )
+                                .toList()
+                )
+                .status(TransactionStatusDto.fromValue(transactionDocument.getStatus().toString()));
+
+    }
+
+    private TransactionInfoDto buildTransactionInfoDto(BaseTransaction baseTransaction) {
+        return new TransactionInfoDto()
+                .transactionId(baseTransaction.getTransactionId().value().toString())
+                .payments(
+                        baseTransaction.getPaymentNotices()
+                                .stream().map(
+                                        paymentNotice -> new PaymentInfoDto()
+                                                .amount(paymentNotice.transactionAmount().value())
+                                                .reason(paymentNotice.transactionDescription().value())
+                                                .paymentToken(paymentNotice.paymentToken().value())
+                                                .rptId(paymentNotice.rptId().value())
+                                ).toList()
+                )
+                .status(TransactionStatusDto.fromValue(baseTransaction.getStatus().toString()));
+
+    }
+
+    private Mono<Boolean> wasTransactionAuthorized(
+                                                   TransactionId transactionId
+    ) {
+        /*
+         * @formatter:off
+         * Searching the transaction with authorization completed event returning a Mono<Boolean>
+         * reflecting the fact that transaction have been previously authorized or not.
+         * The check is performed directly on the authorization
+         * completed event and not on the fact that the reduced transaction is instance
+         * of BaseTransactionWithCompletedAuthorization because the transaction can go
+         * in REFUND or EXPIRED statuses and their aggregates does not directly extend
+         * the BaseTransactionWithCompletedAuthorization just because transaction can
+         * EXPIRE in any state and can be REFUND also if the authorization was only
+         * requested so those aggregates. So a check is performed against the
+         * TRANSACTION_AUTHORIZATION_COMPLETED_EVENT event
+         * @formatter:on
+         */
```

Comment:
```suggestion
        /*
         * @formatter:off
         *
         * This method determines whether transaction has been previously authorized or not 
         * by searching for an authorization completed event.
         * The check is performed directly on the presence of an authorization completed event
         * and not on the fact that the transaction aggregate is an instance of `BaseTransactionWithCompletedAuthorization`
         * because a generic transaction can go in the REFUNDED or EXPIRED states without undergoing authorization
         * (the corresponding aggregates do not extend, in fact, `BaseTransactionWithCompletedAuthorization`).
         * 
         * This can happen, for example, when a transaction expires before getting a payment gateway response
         * (for the EXPIRED state; if in REFUNDED that means the transaction was already refunded).
         *
         * @formatter:on
         */
```